src/tsickle.ts

```diff
@@ -566,7 +578,44 @@ class Annotator extends ClosureRewriter {
 
   annotate() {
     this.visit(this.file);
-    return this.getOutput();
+    const out = super.getOutput();
+    out.output = this.extraImports + out.output;
+    return out;
+  }
+
+  ensureSymbolImported(sym: ts.Symbol) {
+    // TODO(martinprobst): it's unclear when a symbol wouldn't have a declaration, maybe just for
+    // some builtins (e.g. Symbol)?
+    if (!sym.declarations || sym.declarations.length === 0) return;
+    // A symbol declared in this file does not need to be imported.
+    if (sym.declarations.some(d => d.getSourceFile() === this.file)) return;
+    // Find an exported declaration.
+    // Because tsickle runs with the --declaration flag, all types referenced from exported types
+    // must be exported, too, so there must either be some declaration that is exported, or the
+    // symbol is actually a global declaration (declared in a script file, not a module).
+    const decl = sym.declarations.find(d => hasModifierFlag(d, ts.ModifierFlags.ExportDefault));
+    if (!decl) return;
+    // Exclude symbols declared in `declare global {...}` blocks, they are glocal and don't need
+    // imports.
+    let current: ts.Node|undefined = decl;
+    while (current) {
+      if (current.flags & ts.NodeFlags.GlobalAugmentation) return;
+      current = current.parent;
+    }
+    // Actually import the symbol.
+    const sf = decl.getSourceFile();
+    const fn = sf.fileName;
```

Comment:
Only used once and a bit far away, might be worth just inlining it.

Reply:
Done.